geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/records/GeoAvroRecordSetWriterFactory.scala

```diff
@@ -0,0 +1,111 @@
+import org.locationtech.jts.geom.Point
+import org.opengis.feature.simple.SimpleFeatureType
+
+@Tags(Array("avro", "geoavro", "result", "set", "recordset", "record", "writer", "serializer", "row"))
+@CapabilityDescription("Writes the contents of a RecordSet as GeoAvro which AvroToPutGeoMesa* Processors can use.")
+class GeoAvroRecordSetWriterFactory extends AbstractControllerService with RecordSetWriterFactory {
+  // NB: This is the same as what the InheritSchemaFromRecord Strategy does
+  override def getSchema(map: util.Map[String, String], recordSchema: RecordSchema): RecordSchema = recordSchema
+
+  override def createWriter(componentLog: ComponentLog,
+                            recordSchema: RecordSchema,
+                            outputStream: OutputStream,
+                            map: util.Map[String, String]): GeoAvroRecordSetWriter = {
+    new GeoAvroRecordSetWriter(componentLog, recordSchema, outputStream, getConfigurationContext.getProperties)
+  }
+
+  override def getSupportedPropertyDescriptors: util.List[PropertyDescriptor] =
+    Collections.singletonList(GEOMETRY_COLUMNS)
+}
+
+object GeoAvroRecordSetWriterFactory {
+  val GEOMETRY_COLUMNS: PropertyDescriptor = new PropertyDescriptor.Builder()
+    .name("Geometry Columns")
+    .description("Comma-separated list of columns with geometries with type " +
+      "(and optionally the format, (Wkt, Wkb)).  " +
+      "Example: position:Point,line:LineString.  " +
+      "Note the first field is used as a the default geometry.")
+    .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .required(false).build
+
+  val TYPE_NAME: PropertyDescriptor = new PropertyDescriptor.Builder()
+    .name("SimpleFeature TypeName")
+    .description("The type name for the output.")
+    .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .required(false).build
+}
+
+class GeoAvroRecordSetWriter(componentLog: ComponentLog, recordSchema: RecordSchema, outputStream: OutputStream, map: util.Map[PropertyDescriptor, String]) extends AbstractRecordSetWriter(outputStream) {
+  private val schemaAccessWriter = new SchemaNameAsAttribute()
+  // Wkt is assumed to be the default GeometryEncoding.
+  // TODO:  Extract this as a property?
+  private val encodings = getEncodings(map, GeometryEncoding.Wkt)
+  private val defaultGeometryColumn: Option[String] = Option(map.get(GEOMETRY_COLUMNS)).map(_.split(":")(0))
+  val typeName: Option[String] = Option(map.get(TYPE_NAME))
+
+  val geometryColumns: Seq[GeometryColumn] = encodings.map { case (k, v) =>
+    GeometryColumn(k, v.clazz, defaultGeometryColumn.isDefined && defaultGeometryColumn.get.equals(k))
+  }.toSeq
+
+  val recordConverterOptions: RecordConverterOptions = RecordConverterOptions(typeName, None, geometryColumns)
+  lazy val converter: SimpleFeatureRecordConverter =
```

Comment:
making this lazy isn't getting anything since you reference it with `private val sft` below. does it need to be lazy? if so, should make all the dependent vals lazy too

Reply:
Dropping the lazy's.